app/controllers/hosts_controller.rb

```diff
@@ -434,11 +434,10 @@ def update_multiple_hostgroup
       redirect_to(select_multiple_hostgroup_hosts_path)
       return
     end
-    hg = Hostgroup.find_by_id(id)
     # update the hosts
     @hosts.each do |host|
-      host.hostgroup = hg
-      host.save(:validate => false)
+      attributes = host.apply_inherited_attributes("hostgroup_id" => id)
+      host.update(attributes)
```

Comment:
Some concerns / questions:

1. Without `host.hostgroup = hg`, isn't this applying attributes from the new hostgroup without actually assigning that hostgroup to the host?

2. Without `:validate => false`, I'm concerned this `.update` may fail sometimes and just return false, resulting in a non-updated host.

3. Should _all_ inherited attributes be applied when editing a hostgroup? When the host already has values for the inherited fields, are we handling all those properly and consistently? Is it correct to have the hostgroup inheritance override fields, even if they've been changed for a host? Would be good to get input from some well-versed Foreman folks. @ares @stejskalleos ?

4. This change is in the controller, which only comes into play when an API endpoint is hit. Shouldn't something change in the model, instead?
5. Should this code be in Katello and not Foreman, since it deals with changing content view and lifecycle environment? I could see doing it in `host_managed_extensions` as a `before_update`, maybe. (This would also satisfy the item above)

Reply:
@jeremylenz Thanks for reviewing this.

> 1. Without `host.hostgroup = hg`, isn't this applying attributes from the new hostgroup without actually assigning that hostgroup to the host?

`host.apply_inherited_attributes("hostgroup_id" => id)` returns `"hostgroup_id" => id + inherited_attributes` in a hash. So the new hostgroup does not get lost.

> 2. Without `:validate => false`, I'm concerned this `.update` may fail sometimes and just return false, resulting in a non-updated host.

It does not make sense to me why `:validate => false` is required there when updating a host's hostgroup.
In this PR, I'm copying the behavior of `Edit host` to update a host group which is handled by [this code](https://github.com/theforeman/foreman/blob/a3dce620805d5f3c292a07fc747b2fbf14523cf8/app/controllers/hosts_controller.rb#L113).

> 3. Should _all_ inherited attributes be applied when editing a hostgroup? When the host already has values for the inherited fields, are we handling all those properly and consistently? Is it correct to have the hostgroup inheritance override fields, even if they've been changed for a host? Would be good to get input from some well-versed Foreman folks. @ares @stejskalleos ?

_all_ inherited attributes may look like `"puppet_proxy_id"=>nil, "puppet_ca_proxy_id"=>nil, "compute_profile_id"=>nil, "realm_id"=>nil, "compute_resource_id"=>nil, "content_facet_attributes"=>{"kickstart_repository_id"=>nil, "content_view_id"=>2, "lifecycle_environment_id"=>2, "content_source_id"=>nil}`. 
I would expect most of them with `nil` for `Change Group` button.

> 4. This change is in the controller, which only comes into play when an API endpoint is hit. Shouldn't something change in the model, instead?

Not sure what the issue is here.

> 5. Should this code be in Katello and not Foreman, since it deals with changing content view and lifecycle environment? I could see doing it in `host_managed_extensions` as a `before_update`, maybe. (This would also satisfy the item above)

I had the same debate as to where the change should locate. Hostgroup is a Foreman concept while LCE/CV delongs to Katello. But `Change Group` is handled by Foreman's `/hosts/update_multiple_hostgroup`.